message/router.go

```diff
@@ -185,7 +200,7 @@ func (r *Router) AddHandler(
 	publishTopic string,
 	publisher Publisher,
 	handlerFunc HandlerFunc,
-) {
+) *handler {
```

Comment:
`handler` is an unexported struct right now, so it might be better to not return it outside of the package (it's only for internal use right now). Instead, I would suggest to add new exported struct, that will hold only handler and router and will have a single public method `AddMiddleware`.